big-o-notation/big-o.js

```diff
@@ -0,0 +1,44 @@
+/* exported uniqueQuadratic, uniqueLinear */
+
+function uniqueLinear(words) {
+  const seen = {};                  // 1 * 1 = O(1)
+  const unique = [];                // 1 * 1 = O(1)
+  for (
+    let i = 0;                      // 1 * 1 = O(1)
+    i < words.length;               // 2 * n = O(2n)
+    i++                             // 2 * n = O(2n)
+  ) {
+    const word = words[i];          // 2 * n = O(2n)
+    if (!seen[word]) {              // 3 * n = O(3n)
+      seen[word] = true;
+      unique[unique.length] = word; // 3 * n = O(3n)
+    }
+  }
+  return unique;
+} // Big O Notation for uniqueLinear: O(n)
+
+function uniqueQuadratic(words) {
+  const unique = [];                // 1 * 1 = O(1)
+  for (
+    let i = 0;                      // 1 * 1 = O(1)
+    i < words.length;               // 2 * n = O(2n)
+    i++                             // 2 * n = O(2n)
+  ) {
+    const word = words[i];          // 2 * n = O(2n)
+    let isUnique = true;            // 1 * 1 = O(1)
+    for (
+      let c = 0;                    // 1 * 1 = O(1)
+      c < i;                        // 1 * 1 = O(1)
+      c++                           // 2 * n = O(2n)
+    ) {
+      const comparing = words[c];   // 2 * n = O(2n)
+      if (comparing === word) {     // 2 * n = O(2n)
+        isUnique = false;           // 1 * 1 = O(1)
+      }
+    }
+    if (isUnique) {                 // 1 * 1 = O(1)
+      unique[unique.length] = word; // 2 * n = O(2n)
+    }
```

Comment:
Each of these operations is executed at each index of the input array `n` times, so the `1` worst cases become `n` and the `n` worst cases become `n^2`. Though not always the case, nested loops in which you have to loop through the input array again for every element of the input array means `n^2` complexity.